sanity-check/main.py

```diff
@@ -145,12 +146,39 @@ def get_clinical(metadata,clinical_url,api_token):
 
         sample_ind=[ele for ele,sample in enumerate(response.json()['specimens'][specimen_ind[0]]['samples']) if sample['sampleId']==return_metadata['sample_id']]
         if len(sample_ind)!=1:
-            sys.exit("ID Mismatch detected. Sample_id:'%s'/'%s'  was not found within Specimen:'%s'/'%s' 's samples" % (metadata['submitter_sample_id'],return_metadata['sample_id'],metadata['submitter_specimen_id'],return_metadata['specimen_id'],))
+            sys.exit("ID Mismatch detected. Sample_id:'%s'/'%s'  was not found within Specimen:'%s'/'%s' 's samples" % (metadata['submitter_sample_id'],return_metadata['sample_id'],metadata['submitter_specimen_id'],return_metadata['specimen_id']))
         return_metadata['sample_type']=response.json()['specimens'][specimen_ind[0]]['samples'][sample_ind[0]]['sampleType']
         return_metadata['submitter_sample_id']=response.json()['specimens'][specimen_ind[0]]['samples'][sample_ind[0]]['submitterId']
 
+    if return_metadata['tumour_normal_designation']=="Tumour":
+        #WGS, WXS, RNA-Seq, Bisulfite-Seq, ChIP-Seq, Targeted-Seq
+        if metadata.get("submitter_matched_normal_sample_id"):
+            check_tumour_sample_exists(metadata,response.json())
+        else:
+            if metadata.get("experimental_strategy")!="RNA-Seq" and metadata.get("experimental_strategy")!= "Targeted-Seq":
```

Comment:
Maybe it is better to enforce the existence of matchedNormal for WGS and WXS, and loose it for all the others? like:
```
if metadata.get("experimental_strategy") in ['WGS', 'WXS']:
```
So that we don't need any changes if there are other strategies we accept in the future.

Reply:
Reworked that logic as per suggestion:

https://github.com/icgc-argo/argo-data-submission/blob/sanity-check%400.1.2/sanity-check/main.py#L155-L159

Reply:
@edsu7 , I think the logic is still not very correct yet. Should it be:
```
if `tumour_normal_designation`==`Tumour`:
   if `submitter_matched_normal_sample_id` is populated:
       should do `check_normal_sample_exists` no matter what experimental strategy is
   else
      if `experimental_strategy` in ['WGS', 'WXS']:  error out
else:
   check whether `submitter_matched_normal_sample_id` is null
```

Reply:
Reworked logic:
https://github.com/icgc-argo/argo-data-submission/blob/c52a2d4b6b9c88b650b8ff2f6603362ec760d781/sanity-check/main.py#L155-L159